route.go

```diff
@@ -12,6 +12,10 @@ import (
 	"strings"
 )
 
+var ErrorNoHostOrPath = errors.New("mux: route doesn't have a host or path")
+var ErrorNoHost = errors.New("mux: route doesn't have a host")
+var ErrorNoPath = errors.New("mux: route doesn't have a path")
```

Comment:
What is the use case for allowing the application to distinguish between these errors?  If there's no use case, then it's best to leave them out the exported interface for the package. Less is more.

If these variables are added to the package, then there should be documentation describing where these errors are returned.  Without documentation, these errors are noise.